### ISSUE TITLE
Consolidate API documentation and implementation

### ISSUE THREAD

**Issue by pral2a** [fablabbcn/fablabs.io#577]
Current [API documentation](https://docs.fablabs.io/swagger/) is not working at all. Also most of the [examples](https://docs.fablabs.io/examples/) related to the API are not working neither. Actually, there are [3 API versions](https://github.com/fablabbcn/fablabs.io/tree/17dc98c9b28ce614c964fed783ae8e2b93453942/app/controllers/api) but none of them is properly documented leading to a lot of difficulties to use the API, mostly for externals.

My suggestions are:

- [ ] Asses documentation and API version status
- [ ] Reduce the number of end-points to be supported supported and maintain only one API version
    - [ ] Confirm what end-points are used by "fabcloud" mgmt scripts
    - [ ] Provide basics methods (labs, machines...) with filter capabilities focusing in Rails standards to reduce maintenance efforts
- [ ] Create basic documentation with methods and payload examples
- [ ] Remove unsupported old documentation or clearly indicate as legacy

**Comment by viktorsmari** [comment]
Another thing is that the API versioning functionality is using a deprecated gem called `rocket_pants`

This gem is preventing us to update to Rails 6

It prevented us also to update to Rails 5, but we found a maintained fork of it, which is now also not maintained.

https://github.com/Sutto/rocket_pants <-- last update 2015
https://github.com/parse/rocket_pants <-- last update 2018 (fork of the previous, we are using this one)

**Comment by MacTwister** [comment]
Some input from my side as Viktor shared this with me. 

1. For the Fabacademy student scripts, we are mainly interested in the `users` endpoint (searching and creating user accounts).

> Another thing is that the API versioning functionality is using a deprecated gem called `rocket_pants`

I'm not so bothered if you would like to move the API to fix controller instead of using that gem. (if you guys are sure nobody else uses those APIs). Do you have any usage analytics on API request? Then let's align on the exact changes, to prepare for the next cycle.


2. The OAuth (Doorkeeper) stuff uses the `api.fablabs.io` URL, not sure where/why this comes from. I only know this when setting up Gitlab and other auth stuff to work with Fablabs.io (probably this could move to the `www.` domain without breaking auth flows, but needs testing). 


3. Continuing on the Auth flow, (maybe related to #444 ), the Gitlab and other platforms that login with FablabsIO, will fetch profile information after authorization, using the URL `https://api.fablabs.io/0/me.json`. Not sure if related to the versioning stuff, but so you know to make sure it doesn't break.



4. Seems like a good idea to switch to the basic "get" api stuff for normal users to use the `.json` rails/integrated stuff. Which would be cleaner for those simple tasks (eg. like students doing research or graphs), though I'd review what information you are output ing to make sure no private information is leaking on any of those end points (ie. `users/xxxx.json` does not leak the email).

**Comment by MacTwister** [comment]
After a few years of stagnation and with the API blocking further Rails upgrades, I decided to take the plunge and remove the old `rocket_pants`-based API.

A quick dive into the recent access logs showed only two endpoints with meaningful traffic:

- `GET /0/me.json` (~10k requests, OAuth clients)
- `GET /0/labs.json` (~1k requests, researchers/analysts)

As pointed out in previous posts, the gap between documented and actual API usage made it clear that a fresh start was needed. I’ve moved forward with a new API under `www.fablabs.io/api`. 

I recognize the effort that went into the previous API, and I could have kept the old routes running without the gem. However, with only a handful of routes in use—and little context on their original purpose or users; starting fresh makes maintenance simpler and more sustainable for the future.

I have implemented these **initial routes** to start of with:

- `/api/me` (same controller as before)
- `/api/labs` (same logic as `/0/labs.json`)
- `/api/users` (same logic as before, for academy scripts)

A nice side effect: OAuth now runs on the main `www.` subdomain, so users won’t need to re-login as they did with the old `api.` subdomain.

The old API subdomain will return 404s for all routes (except OAuth, which will remain until clients migrate), allowing us to monitor lingering traffic. The previous `/2/` API followed the [json:api](https://jsonapi.org/) spec—something we can revisit if desired.

This change unblocks the Rails upgrade and gives us a clean foundation. Feedback and ideas are welcome—let’s build from here!

**Comment by zeno4ever** [comment]
FYI I use the endpoint https://api.fablabs.io/0/labs.json for my [mapall.space](https://mapsall.space) website. 

My aim is to have a list of **active** fablabs and show this on a worldwide map. This kind of works only problem (but this is not a software problem) is that some fablabs are closed but still on the list. As far I know there is no active checking if a space is still open. 

If you have any questions about my usage, just let me know!

Regards, 
Dave

**Comment by MacTwister** [comment]
Hi @zeno4ever,

Thanks for sharing your use case and the context around mapall, nice project! Reading your message actually made me reconsider the plan to redirect the old `/0/labs.json` endpoint. I’ll keep it active for a while longer, even though most HTTP clients handle redirects just fine.

This will give us more time to brainstorm if there’s interest in changing how the labs endpoint works (e.g., response structure, pagination, filtering), or if we should keep the same logic on the new URL. I am going to focus on the rest of the upgrades first.

How to handle closed labs is an interesting discussion—something for a new issue to explore that further.

Cheers!
